test/e2e/framework/deployment.go

```diff
@@ -129,6 +129,58 @@ func (f *Framework) NewEchoDeployment(opts ...func(*deploymentOptions)) {
 	assert.Nil(ginkgo.GinkgoT(), err, "waiting for endpoints to become ready")
 }
 
+// NewHttpbunDeployment creates a new single replica deployment of the httpbun
+// server image in a particular namespace we return the ip for testing purposes
+func (f *Framework) NewHttpbunDeployment(opts ...func(*deploymentOptions)) string {
+	options := &deploymentOptions{
+		namespace: f.Namespace,
+		name:      HTTPBunService,
+		replicas:  1,
+	}
+	for _, o := range opts {
+		o(options)
+	}
+
+	//TODO: once we have the httpbun image in our registry we should swap the
+	//image used here
+	deployment := newDeployment(options.name, options.namespace, "ghcr.io/sharat87/httpbun", 80, int32(options.replicas),
```

Comment:
here, if we have the internal image built to be used, or if we can use the image on registry.k8s.io would be better :D This is why we would need first the httpbun image creation and promotion on another PR, then we can just use it here

cc @longwuyuan